src/composables/useCamera/index.ts

```diff
@@ -1,17 +1,18 @@
-import { computed, watchEffect, onUnmounted, ref } from 'vue'
 import { Camera, OrthographicCamera, PerspectiveCamera } from 'three'
+import { computed, triggerRef, watch, watchEffect } from 'vue'
 
-import type { TresScene } from '../../types'
 import type { TresContext } from '../useTresContextProvider'
 
 
-export const useCamera = ({ sizes, scene }: Pick<TresContext, 'sizes'> & { scene: TresScene }) => {
-
-  // the computed does not trigger, when for example the camera postion changes
-  const cameras = ref<Camera[]>([])
+export const useCamera = ({ sizes, scene }: Pick<TresContext, 'sizes' | 'scene'>) => {
+  const cameras = computed<Camera[]>(() => scene.value.children.filter((i): i is Camera => (i as any).isCamera))
   const camera = computed<Camera | undefined>(
-    () => cameras.value[0]
+    () => cameras.value[0],
   )
+  // don't known why need manually trigger here
+  watch(camera, () => {
+    triggerRef(cameras)
+  }, { deep: true })
```

Comment:
just use `ShallowRef<Scene>` and it will track by manually `triggerRef`, so I refactor composables about using `scene`

***
but no idea about L13-15, if remove the camera should always be undefined, but this stuff doesn't make sense :P

Reply:
@enpitsuLin  scene is widely used in `cientos` , would the use of ShallowRef affect the reactivity of the abstractions?